### DISCUSSION TITLE
Events and listeners in v3 vs v2

### DISCUSSION THREAD

**Discussion by shamil8124** [livewire/livewire#8590] [1. Help]
Hi everyone.

In v2, if I had nested livewire components that were rendered inside a blade if condition, and they were listening for events from the parent component or other sibling components, everything would work perfectly. 

In v3, it looks like any of these components inside an @if condition no longer work correctly. 

Is there a way to default back to v2 behaviour?

**Comment by shamil8124** [comment]
I have a feeling this is probably due to the fact that dispatch() is now a "browser event" and we're no longer using "emit." So the actual HTML needs to be on the page for the event to register with the listener.

**Comment by shamil8124** [comment]
Another gotcha with the new events system. 

Since everything passes through the browser, if you're dispatching a few thousand records in an array, listening for it, and then hydrating another property, your app will freeze up. 

v2 did this server side. 

Wonder what the recommended approach would be to refactor those bits?

**Reply by joshhanley** [reply]
@shamil8124 all events V2 and V3 went through the browser. The only difference is V2 there was a custom event bus, where as V3 uses native browser events.

So I'd likely say something else is going on. But I wouldn't be dispatching an event with thousands of records at most I would just include IDs.
